File: app/utils/jwt.py
from jose import jwt, JWTError
from datetime import datetime, timedelta

SECRET_KEY = "super-secret"  # change for prod
ALGORITHM = "HS256"
EXPIRE_MINUTES = 60 * 24  # 1 day

def create_token(player_id: str, name: str) -> str:
    expire = datetime.utcnow() + timedelta(minutes=EXPIRE_MINUTES)
    payload = {
        "sub": player_id,
        "name": name,
        "exp": expire
    }
    return jwt.encode(payload, SECRET_KEY, algorithm=ALGORITHM)

def verify_token(token: str) -> dict:
    try:
        payload = jwt.decode(token, SECRET_KEY, algorithms=[ALGORITHM])
        return payload  # contains player_id in "sub"
    except JWTError:
        return None

File: server/src/utils/jwt.js
const jwt = require("jsonwebtoken");
const { SECRET_KEY, EXPIRE_MINUTES } = require("../constants");

/**
 * Create a JWT token for a player
 * @param {string} playerId - The player's ID
 * @param {string} name - The player's name
 * @returns {string} JWT token
 */
function createToken(playerId, name) {
  const payload = {
    sub: playerId,
    name,
    exp: Math.floor(Date.now() / 1000) + EXPIRE_MINUTES * 60,
  };
  return jwt.sign(payload, SECRET_KEY);
}

/**
 * Verify and decode a JWT token
 * @param {string} token - JWT token
 * @returns {Object|null} Decoded payload or null if invalid
 */
function verifyToken(token) {
  try {
    return jwt.verify(token, SECRET_KEY);
  } catch (error) {
    return null;
  }
}

module.exports = {
  createToken,
  verifyToken,
};
